h2o-persist-s3/src/test/java/water/persist/PersistS3Test.java

```diff
@@ -9,17 +9,21 @@
 import water.util.FileUtils;
 
 import java.net.URI;
+import java.util.ArrayList;
 
 import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
 
 /**
  * Created by tomas on 6/27/16.
  */
 public class PersistS3Test extends TestUtil {
-
+  private static final String AWS_ACCESS_KEY_PROPERTY_NAME = "AWS_ACCESS_KEY_ID";
```

Comment:
The names of these properties coincide with the names AWS libraries are using to lookup the credentials.

That is why test `testS3UriToKeyUriCredentials` doesn't actually prove that credentials encoded in URL will work (the S3 libraries might take it from the system properties). Aka if I intentionally remove the credentials the test will still pass.